internal/controllers/clusterrequest_controller.go

```diff
@@ -1902,59 +1901,83 @@ func (t *clusterRequestReconcilerTask) updateNodeStatusWithHostname(ctx context.
-				utils.CRconditionTypes.HardwareProvisioned,
-				utils.ConditionReason(provisionedCondition.Reason),
-				provisionedCondition.Status,
-				provisionedCondition.Message)
-		} else {
-			utils.SetStatusCondition(&t.object.Status.Conditions,
-				utils.CRconditionTypes.HardwareProvisioned,
-				utils.CRconditionReasons.Unknown,
-				metav1.ConditionUnknown,
-				"Unknown state of hardware provisioning",
-			)
-		}
+	if t.object.Status.NodePoolRef == nil {
+		t.object.Status.NodePoolRef = &oranv1alpha1.NodePoolRef{}
+	}
 
-		if err := utils.UpdateK8sCRStatus(ctx, t.client, t.object); err != nil {
-			t.logger.ErrorContext(
+	t.object.Status.NodePoolRef.Name = nodePool.GetName()
+	t.object.Status.NodePoolRef.Namespace = nodePool.GetNamespace()
+	if t.object.Status.NodePoolRef.HardwareProvisioningCheckStart.IsZero() {
+		t.object.Status.NodePoolRef.HardwareProvisioningCheckStart = metav1.Now()
+	}
+
+	provisionedCondition := meta.FindStatusCondition(
+		nodePool.Status.Conditions, string(hwv1alpha1.Provisioned))
+	if provisionedCondition != nil {
+		status = provisionedCondition.Status
+		reason = provisionedCondition.Reason
+		message = provisionedCondition.Message
+
+		if provisionedCondition.Status == metav1.ConditionFalse && reason == string(hwv1alpha1.Failed) {
+			t.logger.InfoContext(
 				ctx,
-				"Failed to update the HardwareProvisioning status for ClusterRequest",
-				slog.String("name", t.object.Name),
-				slog.Any("specificError", err),
+				fmt.Sprintf(
+					"NodePool %s in the namespace %s provisioning failed",
+					nodePool.GetName(),
+					nodePool.GetNamespace(),
+				),
 			)
-			return fmt.Errorf("failed to update HardwareProvisioning status: %w", err)
+			// Ensure a consistent message for the cluster request, regardless of which plugin is used.
+			message = "Hardware provisioning failed"
+			timedOutOrFailed = true
 		}
-	} else if nodePool.ObjectMeta.Namespace == utils.TempDellPluginNamespace || nodePool.ObjectMeta.Namespace == utils.UnitTestHwmgrNamespace {
-		// TODO: For test purposes only. Code to be removed once hwmgr plugin(s) are fully utilized
-		meta.SetStatusCondition(
-			&nodePool.Status.Conditions,
-			metav1.Condition{
-				Type:   string(hwv1alpha1.Unknown),
-				Status: metav1.ConditionUnknown,
-				Reason: string(hwv1alpha1.NotInitialized),
-			},
-		)
-		if err := utils.UpdateK8sCRStatus(ctx, t.client, nodePool); err != nil {
-			t.logger.ErrorContext(
+	} else {
+		// No provisioning condition found, set the status to unknown.
+		status = metav1.ConditionUnknown
+		reason = string(utils.CRconditionReasons.Unknown)
+		message = "Unknown state of hardware provisioning"
+	}
+
+	// Check for timeout if not already failed or provisioned
+	if status != metav1.ConditionTrue && reason != string(hwv1alpha1.Failed) {
+		elapsedTime := time.Since(t.object.Status.NodePoolRef.HardwareProvisioningCheckStart.Time)
+		if elapsedTime >= time.Duration(t.object.Spec.Timeout.HardwareProvisioning)*time.Second {
```

Comment:
Should this be `time.Minute`?

Reply:
Yes, just realize that it is in minutes, I saw the default 90 and assumed it was in seconds.

Reply:
Fixed.

Reply:
Yes.